### DISCUSSION TITLE
Autogenerate .reality from glb, not usdz

### DISCUSSION THREAD

**Discussion by GianiNavette** [google/model-viewer#3858] [Ideas]
It would be cool to be able to have .reality files.
One key point is anchor placement.
- in a reality file you can specify the anchor, which is impossible with usdz.

Would be possible to reverse engineer .reality files.
I'll be happy to help, i own a mac and can provide any type of file you want.

**Comment by elalish** [comment]
PRs welcome if you'd like to create a .reality exporter/converter! Meanwhile, if you already created a .reality file, you can set that as the `ios-src` and it will work as expected on iOS QuickLook.

**Reply by GianiNavette** [reply]
> PRs welcome if you'd like to create a .reality exporter/converter! Meanwhile, if you already created a .reality file, you can set that as the `ios-src` and it will work as expected on iOS QuickLook.

Yes would love to contribule, I do know about the use of ios-src, but i would like to be able to autogenerate it from glb/gltf files, do you have any insight about reverse engineer .reality files, i already unzipped an .reality files but there was lot of information to digest, do you know how three js community found a way to do it with usdz ?
Thanks 👍

**Reply by elalish** [reply]
Three js did it with lots of hard work reading and understanding those files. Here's the docs on ios-src (which does handle both .reality and .usdz) https://modelviewer.dev/docs/#entrydocs-augmentedreality-attributes-iosSrc

I'm not sure if auto-conversion is really what you want; where would you get the AR anchor info from? It's not part of the glTF. Unless I misunderstand...

**Reply by GianiNavette** [reply]
> Three js did it with lots of hard work reading and understanding those files. Here's the docs on ios-src (which does handle both .reality and .usdz) https://modelviewer.dev/docs/#entrydocs-augmentedreality-attributes-iosSrc
> 
> I'm not sure if auto-conversion is really what you want; where would you get the AR anchor info from? It's not part of the glTF. Unless I misunderstand...

I saw that inside the file ar.ts, ar-placement="wall" has no impact on the usdz generation.
```
   if (srcUrl.hash) {
          modelUrl.hash = srcUrl.hash;
        }
      }

      if (this.arScale === 'fixed') {
        if (modelUrl.hash) {
          modelUrl.hash += '&';
        }
        modelUrl.hash += 'allowsContentScaling=0';
      }
```
So i did my dinging and I saw in Apple docs there was no way to choose the anchor of the usdz only an reality file would allow that.

I think creating an exporter from glb to .reality would allow the use of anchor placement present in reality composer
![image](https://user-images.githubusercontent.com/71259828/194359092-7d7e23ad-9b1c-458b-bd2a-21c9fa83d81e.png)
as this image show what anchor exist on ios.
